common/txmgr/types/head_view.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import "github.com/ethereum/go-ethereum/common"
+
+// HeadView provides minimal access to a chain's head, as needed by the TxManager.
+// This is a generic interface which ALL chains will implement.
+//
+//go:generate mockery --quiet --name HeadView --output ./mocks/ --case=underscore
+type HeadView[HEAD any] interface {
```

Comment:
What is the significance of `View` here? These types of modifiers are usually a code smell in Go.
```suggestion
type Head[HEAD any] interface {
```

Reply:
Ya, I couldn't come up with a better name. Head was just used everywhere in EVM so tried to not use it.
I'm ok to do Head. 
Basically Head[Native any]. That sound good?